code/src/beetle.cpp

```diff
@@ -0,0 +1,75 @@
+#include <beetle.hpp>
+
+namespace r2d2::moving_platform {
+
+    beetle_c::beetle_c(r2d2::usart::usart_ports &usart_port, unsigned int baud_rate,
+             hwlib::pin_out &reset_pin, 
+             base_comm_c &comm):
+             moving_platform_c(comm),
+             qik_2s12v10_motorcontroller (usart_port, baud_rate, reset_pin)
+    {
+        
+        qik_2s12v10_motorcontroller.init();
+        qik_2s12v10_motorcontroller.get_configuration_parameter(1);
+
+    }
+    void beetle_c::set_speed(const int8_t &new_speed){
+        // Todo: depending on the motor configuration, 127 may not be the
+        // maximum value
+        // Todo: change the speed variable to the actual speed.
+
+        // the speed is given in percentage (-100 (backward) till 100 (foreward))
+        speed = new_speed;
+        // do not go out of range
+        if(speed > 100){
```

Comment:
missing spaces should be `if (speed > 100) {`

Reply:
Your commit has fixed this